### ISSUE TITLE
Can I download big image without crash

### ISSUE THREAD

**Issue by smolskyaleksey** [hyperoslo/Imaginary#90]
If I download big image via

```swift
self.session.dataTask(with: request, completionHandler: { [weak self] data, response, error}
```

image stored inside response. And I can catch crash. How can I use URLSession's method
```swift
func downloadTask(with request: URLRequest, 
completionHandler: @escaping (URL?, URLResponse?, Error?) -> Void) -> URLSessionDownloadTask
```
to store image in file

**Comment by smolskyaleksey** [comment]
Why do you close issue without any answer?

**Comment by 3lvis** [comment]
Hello @smolskyaleksey,

Sorry for not writing before. I haven't been able to reproduce this issue. Could you help me with additional steps to reproduce? How big was the file you were trying to download? Is this still an issue for you?

**Comment by smolskyaleksey** [comment]
Yes, try to download big image above 100 mgb. dataTask represent image in memory in data parameter. U can crash because you don't have enough memory.

**Comment by 3lvis** [comment]
OK! Will try to reproduce, this might be caused because of limitations in URLSession so it might require us to implement background downloads in order to make this work. WIll investigate further.

**Comment by smolskyaleksey** [comment]
> OK! Will try to reproduce, this might be caused because of limitations in URLSession so it might require us to implement background downloads in order to make this work. WIll investigate further.

You can use 
``` swift
func downloadTask(with request: URLRequest, 
completionHandler: @escaping (URL?, URLResponse?, Error?) -> Void) -> URLSessionDownloadTask
```

It downloads file directly in file system
